Warnings when using clang

There are warnings when using CLANG

```
2021-11-12T15:59:28.4241984Z [  1%] Building C object src/common/CMakeFiles/common.dir/postgres_connection.c.o
2021-11-12T15:59:28.4256696Z [  1%] Building C object src/dominator/CMakeFiles/dominator.dir/lengauerTarjanDominatorTree.c.o
2021-11-12T15:59:28.4274325Z [  1%] Building C object src/dijkstra/CMakeFiles/dijkstra.dir/dijkstra.c.o
2021-11-12T15:59:28.4299017Z [  1%] Building C object src/allpairs/CMakeFiles/allpairs.dir/floydWarshall.c.o
2021-11-12T15:59:28.8282088Z [  2%] Building CXX object src/dominator/CMakeFiles/dominator.dir/lengauerTarjanDominatorTree_driver.cpp.o
2021-11-12T15:59:28.9218818Z [  3%] Building C object src/dijkstra/CMakeFiles/dijkstra.dir/dijkstraVia.c.o
2021-11-12T15:59:28.9237348Z [  3%] Building C object src/common/CMakeFiles/common.dir/e_report.c.o
2021-11-12T15:59:28.9257704Z [  4%] Building CXX object src/allpairs/CMakeFiles/allpairs.dir/floydWarshall_driver.cpp.o
2021-11-12T15:59:29.0620151Z [  5%] Building CXX object src/common/CMakeFiles/common.dir/basePath_SSEC.cpp.o
2021-11-12T15:59:29.1139425Z [  5%] Building CXX object src/dijkstra/CMakeFiles/dijkstra.dir/dijkstra_driver.cpp.o
2021-11-12T15:59:35.0536674Z In file included from /home/runner/work/pgrouting/pgrouting/src/dominator/lengauerTarjanDominatorTree_driver.cpp:44:
2021-11-12T15:59:35.0540451Z /home/runner/work/pgrouting/pgrouting/include/dominator/pgr_lengauerTarjanDominatorTree_driver.hpp:85:40: warning: suggest braces around initialization of subobject [-Wmissing-braces]
2021-11-12T15:59:35.0542611Z                     results.push_back({vid, (idoms[*v] != -1 ? (idoms[*v]+1) : 0) });
2021-11-12T15:59:35.0544421Z                                        ^~~
2021-11-12T15:59:35.0545195Z                                        {  }
2021-11-12T15:59:35.0549828Z /home/runner/work/pgrouting/pgrouting/src/dominator/lengauerTarjanDominatorTree_driver.cpp:82:30: note: in instantiation of member function 'pgrouting::functions::Pgr_LTDTree<pgrouting::graph::Pgr_base_graph<boost::adjacency_list<boost::vecS, boost::vecS, boost::bidirectionalS, pgrouting::Basic_vertex, pgrouting::Basic_edge, boost::no_property, boost::listS>, pgrouting::Basic_vertex, pgrouting::Basic_edge> >::pgr_ltdtree' requested here
2021-11-12T15:59:35.0626169Z         results = fn_LTDTree.pgr_ltdtree(digraph, root_vertex);
2021-11-12T15:59:35.0626800Z                              ^
2021-11-12T15:59:35.0629932Z In file included from /home/runner/work/pgrouting/pgrouting/src/dominator/lengauerTarjanDominatorTree_driver.cpp:44:
2021-11-12T15:59:35.0632260Z /home/runner/work/pgrouting/pgrouting/include/dominator/pgr_lengauerTarjanDominatorTree_driver.hpp:85:45: warning: suggest braces around initialization of subobject [-Wmissing-braces]
2021-11-12T15:59:35.0636374Z                     results.push_back({vid, (idoms[*v] != -1 ? (idoms[*v]+1) : 0) });
2021-11-12T15:59:35.0636940Z                                             ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
2021-11-12T15:59:35.0637373Z                                             {                                    }
2021-11-12T15:59:35.6815976Z [  5%] Building C object src/common/CMakeFiles/common.dir/points_input.c.o
2021-11-12T15:59:35.9893046Z [  6%] Building C object src/common/CMakeFiles/common.dir/matrixRows_input.c.o
2021-11-12T15:59:36.3045808Z [  6%] Building C object src/common/CMakeFiles/common.dir/get_check_data.c.o
2021-11-12T15:59:36.5486253Z [  6%] Building C object src/allpairs/CMakeFiles/allpairs.dir/johnson.c.o
2021-11-12T15:59:36.6126671Z [  7%] Building C object src/common/CMakeFiles/common.dir/edges_input.c.o
2021-11-12T15:59:36.8092945Z [  8%] Building CXX object src/allpairs/CMakeFiles/allpairs.dir/johnson_driver.cpp.o
2021-11-12T15:59:36.9248487Z 2 warnings generated.

2021-11-12T16:00:10.0143216Z /home/runner/work/pgrouting/pgrouting/src/bdAstar/bdAstar_driver.cpp:111:37: warning: suggest braces around initialization of subobject [-Wmissing-braces]
2021-11-12T16:00:10.0266144Z         II_t_rt previousCombination{0, 0};
2021-11-12T16:00:10.0267373Z                                     ^
2021-11-12T16:00:10.0268254Z                                     {}
2021-11-12T16:00:10.0270151Z /home/runner/work/pgrouting/pgrouting/src/bdAstar/bdAstar_driver.cpp:111:40: warning: suggest braces around initialization of subobject [-Wmissing-braces]
2021-11-12T16:00:10.0271517Z         II_t_rt previousCombination{0, 0};
2021-11-12T16:00:10.0272208Z                                        ^
2021-11-12T16:00:10.0273323Z                                        {}
2021-11-12T16:00:10.3606846Z [ 33%] Building CXX object src/trsp/CMakeFiles/trsp.dir/GraphDefinition.cpp.o
2021-11-12T16:00:10.8028599Z /home/runner/work/pgrouting/pgrouting/src/bdAstar/bdAstar_driver.cpp:111:37: warning: suggest braces around initialization of subobject [-Wmissing-braces]
2021-11-12T16:00:10.8029903Z         II_t_rt previousCombination{0, 0};
2021-11-12T16:00:10.8030435Z                                     ^
2021-11-12T16:00:10.8030836Z                                     {}
2021-11-12T16:00:10.8034306Z /home/runner/work/pgrouting/pgrouting/src/bdAstar/bdAstar_driver.cpp:196:21: note: in instantiation of function template specialization 'pgr_bdAstar<pgrouting::graph::Pgr_base_graph<boost::adjacency_list<boost::listS, boost::vecS, boost::bidirectionalS, pgrouting::XY_vertex, pgrouting::Basic_edge, boost::no_property, boost::listS>, pgrouting::XY_vertex, pgrouting::Basic_edge> >' requested here
2021-11-12T16:00:10.8036280Z             paths = pgr_bdAstar(digraph,
2021-11-12T16:00:10.8036789Z                     ^
2021-11-12T16:00:10.8038073Z /home/runner/work/pgrouting/pgrouting/src/bdAstar/bdAstar_driver.cpp:111:40: warning: suggest braces around initialization of subobject [-Wmissing-braces]
2021-11-12T16:00:10.8039197Z         II_t_rt previousCombination{0, 0};
2021-11-12T16:00:10.8039711Z                                        ^
2021-11-12T16:00:10.8040105Z                                        {}
2021-11-12T16:00:11.3196352Z /home/runner/work/pgrouting/pgrouting/src/bdAstar/bdAstar_driver.cpp:111:37: warning: suggest braces around initialization of subobject [-Wmissing-braces]
2021-11-12T16:00:11.3233389Z         II_t_rt previousCombination{0, 0};
2021-11-12T16:00:11.3253788Z                                     ^
2021-11-12T16:00:11.3254811Z                                     {}
2021-11-12T16:00:11.3259089Z /home/runner/work/pgrouting/pgrouting/src/bdAstar/bdAstar_driver.cpp:212:21: note: in instantiation of function template specialization 'pgr_bdAstar<pgrouting::graph::Pgr_base_graph<boost::adjacency_list<boost::listS, boost::vecS, boost::undirectedS, pgrouting::XY_vertex, pgrouting::Basic_edge, boost::no_property, boost::listS>, pgrouting::XY_vertex, pgrouting::Basic_edge> >' requested here
2021-11-12T16:00:11.3268891Z             paths = pgr_bdAstar(
2021-11-12T16:00:11.3269864Z                     ^
2021-11-12T16:00:11.3271618Z /home/runner/work/pgrouting/pgrouting/src/bdAstar/bdAstar_driver.cpp:111:40: warning: suggest braces around initialization of subobject [-Wmissing-braces]
2021-11-12T16:00:11.3273086Z         II_t_rt previousCombination{0, 0};
2021-11-12T16:00:11.3273854Z                                        ^
2021-11-12T16:00:11.3274491Z                                        {}
2021-11-12T16:00:12.7969569Z [ 33%] Building C object src/trsp/CMakeFiles/trsp.dir/new_trsp.c.o
2021-11-12T16:00:12.9807371Z /home/runner/work/pgrouting/pgrouting/src/bdDijkstra/bdDijkstra_driver.cpp:100:37: warning: suggest braces around initialization of subobject [-Wmissing-braces]
2021-11-12T16:00:12.9809500Z         II_t_rt previousCombination{0, 0};
2021-11-12T16:00:12.9810114Z                                     ^
2021-11-12T16:00:12.9810512Z                                     {}
2021-11-12T16:00:12.9868440Z /home/runner/work/pgrouting/pgrouting/src/bdDijkstra/bdDijkstra_driver.cpp:100:40: warning: suggest braces around initialization of subobject [-Wmissing-braces]
2021-11-12T16:00:12.9871400Z         II_t_rt previousCombination{0, 0};
2021-11-12T16:00:12.9871970Z                                        ^
2021-11-12T16:00:12.9872371Z                                        {}
2021-11-12T16:00:13.1143708Z [ 34%] Building CXX object src/trsp/CMakeFiles/trsp.dir/edgeInfo.cpp.o
2021-11-12T16:00:13.7281115Z 6 warnings generated.
2021-11-12T16:00:13.7383634Z [ 34%] Building CXX object src/trsp/CMakeFiles/trsp.dir/pgr_trspHandler.cpp.o
2021-11-12T16:00:13.7811112Z [ 34%] Built target bdAstar
2021-11-12T16:00:13.7863887Z /home/runner/work/pgrouting/pgrouting/src/bdDijkstra/bdDijkstra_driver.cpp:100:37: warning: suggest braces around initialization of subobject [-Wmissing-braces]
2021-11-12T16:00:13.7865744Z         II_t_rt previousCombination{0, 0};
2021-11-12T16:00:13.7866295Z                                     ^
2021-11-12T16:00:13.7892987Z                                     {}
2021-11-12T16:00:13.7897978Z /home/runner/work/pgrouting/pgrouting/src/bdDijkstra/bdDijkstra_driver.cpp:175:21: note: in instantiation of function template specialization 'pgr_bdDijkstra<pgrouting::graph::Pgr_base_graph<boost::adjacency_list<boost::vecS, boost::vecS, boost::bidirectionalS, pgrouting::Basic_vertex, pgrouting::Basic_edge, boost::no_property, boost::listS>, pgrouting::Basic_vertex, pgrouting::Basic_edge> >' requested here
2021-11-12T16:00:13.7901780Z             paths = pgr_bdDijkstra(digraph,
2021-11-12T16:00:13.7902319Z                     ^
2021-11-12T16:00:13.7906224Z /home/runner/work/pgrouting/pgrouting/src/bdDijkstra/bdDijkstra_driver.cpp:100:40: warning: suggest braces around initialization of subobject [-Wmissing-braces]
2021-11-12T16:00:13.7909080Z         II_t_rt previousCombination{0, 0};
2021-11-12T16:00:13.7909635Z                                        ^
2021-11-12T16:00:13.7910038Z                                        {}
2021-11-12T16:00:13.8689374Z [ 35%] Building C object src/max_flow/CMakeFiles/max_flow.dir/max_flow.c.o
2021-11-12T16:00:14.2259482Z [ 35%] Building C object src/max_flow/CMakeFiles/max_flow.dir/minCostMaxFlow.c.o
2021-11-12T16:00:14.2316831Z /home/runner/work/pgrouting/pgrouting/src/bdDijkstra/bdDijkstra_driver.cpp:100:37: warning: suggest braces around initialization of subobject [-Wmissing-braces]
2021-11-12T16:00:14.2318229Z         II_t_rt previousCombination{0, 0};
2021-11-12T16:00:14.2318782Z                                     ^
2021-11-12T16:00:14.2319153Z                                     {}
2021-11-12T16:00:14.2324696Z /home/runner/work/pgrouting/pgrouting/src/bdDijkstra/bdDijkstra_driver.cpp:185:21: note: in instantiation of function template specialization 'pgr_bdDijkstra<pgrouting::graph::Pgr_base_graph<boost::adjacency_list<boost::vecS, boost::vecS, boost::undirectedS, pgrouting::Basic_vertex, pgrouting::Basic_edge, boost::no_property, boost::listS>, pgrouting::Basic_vertex, pgrouting::Basic_edge> >' requested here
2021-11-12T16:00:14.2326762Z             paths = pgr_bdDijkstra(
2021-11-12T16:00:14.2327226Z                     ^
2021-11-12T16:00:14.2332148Z /home/runner/work/pgrouting/pgrouting/src/bdDijkstra/bdDijkstra_driver.cpp:100:40: warning: suggest braces around initialization of subobject [-Wmissing-braces]
2021-11-12T16:00:14.2334761Z         II_t_rt previousCombination{0, 0};
2021-11-12T16:00:14.2336529Z                                        ^
2021-11-12T16:00:14.2338077Z                                        {}
2021-11-12T16:00:14.5858631Z [ 36%] Building C object src/max_flow/CMakeFiles/max_flow.dir/maximum_cardinality_matching.c.o
2021-11-12T16:00:14.9336594Z [ 36%] Building C object src/max_flow/CMakeFiles/max_flow.dir/edge_disjoint_paths.c.o
2021-11-12T16:00:15.2809433Z [ 37%] Building CXX object src/max_flow/CMakeFiles/max_flow.dir/max_flow_driver.cpp.o
2021-11-12T16:00:16.4947152Z 6 warnings generated.
2021-11-12T16:00:52.5027992Z In file included from /home/runner/work/pgrouting/pgrouting/src/components/makeConnected_driver.cpp:39:
2021-11-12T16:00:52.5107326Z /home/runner/work/pgrouting/pgrouting/include/components/pgr_makeConnected.hpp:91:34: warning: suggest braces around initialization of subobject [-Wmissing-braces]
2021-11-12T16:00:52.5109664Z                    results[i] = {src, tgt};
2021-11-12T16:00:52.5110435Z                                  ^~~
2021-11-12T16:00:52.5111076Z                                  {  }
2021-11-12T16:00:52.5113059Z /home/runner/work/pgrouting/pgrouting/include/components/pgr_makeConnected.hpp:91:39: warning: suggest braces around initialization of subobject [-Wmissing-braces]
2021-11-12T16:00:52.5114261Z                    results[i] = {src, tgt};
2021-11-12T16:00:52.5114830Z                                       ^~~
2021-11-12T16:00:52.5115181Z                                       {  }
2021-11-12T16:00:53.1590494Z /home/runner/work/pgrouting/pgrouting/include/components/pgr_makeConnected.hpp:91:34: warning: suggest braces around initialization of subobject [-Wmissing-braces]
2021-11-12T16:00:53.1593342Z                    results[i] = {src, tgt};
2021-11-12T16:00:53.1595351Z                                  ^~~
2021-11-12T16:00:53.1597458Z                                  {  }
2021-11-12T16:00:53.1604046Z /home/runner/work/pgrouting/pgrouting/include/components/pgr_makeConnected.hpp:56:27: note: in instantiation of member function 'pgrouting::functions::Pgr_makeConnected<pgrouting::graph::Pgr_base_graph<boost::adjacency_list<boost::vecS, boost::vecS, boost::undirectedS, pgrouting::Basic_vertex, pgrouting::Basic_edge, boost::no_property, boost::listS>, pgrouting::Basic_vertex, pgrouting::Basic_edge> >::generatemakeConnected' requested here
2021-11-12T16:00:53.1607913Z                    return generatemakeConnected(graph);
2021-11-12T16:00:53.1609668Z                           ^
2021-11-12T16:00:53.1614538Z /home/runner/work/pgrouting/pgrouting/src/components/makeConnected_driver.cpp:72:36: note: in instantiation of member function 'pgrouting::functions::Pgr_makeConnected<pgrouting::graph::Pgr_base_graph<boost::adjacency_list<boost::vecS, boost::vecS, boost::undirectedS, pgrouting::Basic_vertex, pgrouting::Basic_edge, boost::no_property, boost::listS>, pgrouting::Basic_vertex, pgrouting::Basic_edge> >::makeConnected' requested here
2021-11-12T16:00:53.1619931Z         results = fn_makeConnected.makeConnected(undigraph);
2021-11-12T16:00:53.1622304Z                                    ^
2021-11-12T16:00:53.1626683Z In file included from /home/runner/work/pgrouting/pgrouting/src/components/makeConnected_driver.cpp:39:
2021-11-12T16:00:53.1631171Z /home/runner/work/pgrouting/pgrouting/include/components/pgr_makeConnected.hpp:91:39: warning: suggest braces around initialization of subobject [-Wmissing-braces]
2021-11-12T16:00:53.1634180Z                    results[i] = {src, tgt};
2021-11-12T16:00:53.1636739Z                                       ^~~
2021-11-12T16:00:53.1639217Z                                       {  }
2021-11-12T16:00:53.8622119Z [ 63%] Building CXX object src/max_flow/CMakeFiles/max_flow.dir/pgr_flowgraph.cpp.o
2021-11-12T16:00:54.8548368Z 4 warnings generated.
2021-11-12T16:00:54.8888451Z [ 64%] Building CXX object src/components/CMakeFiles/components.dir/componentsResult.cpp.o
2021-11-12T16:00:55.2830647Z /home/runner/work/pgrouting/pgrouting/src/components/componentsResult.cpp:51:32: warning: suggest braces around initialization of subobject [-Wmissing-braces]
2021-11-12T16:00:55.2832405Z             results.push_back({edge_id, component_id});
2021-11-12T16:00:55.2833284Z                                ^~~~~~~
2021-11-12T16:00:55.2833654Z                                {      }
2021-11-12T16:00:55.2835861Z /home/runner/work/pgrouting/pgrouting/src/components/componentsResult.cpp:51:41: warning: suggest braces around initialization of subobject [-Wmissing-braces]
2021-11-12T16:00:55.2837280Z             results.push_back({edge_id, component_id});
2021-11-12T16:00:55.2837788Z                                         ^~~~~~~~~~~~
2021-11-12T16:00:55.2838188Z                                         {           }
2021-11-12T16:00:55.5829294Z 2 warnings generated.
/sequentialVertexColoring_driver.cpp:39:
2021-11-12T16:01:34.6235468Z /home/runner/work/pgrouting/pgrouting/include/coloring/pgr_sequentialVertexColoring.hpp:135:34: warning: suggest braces around initialization of subobject [-Wmissing-braces]
2021-11-12T16:01:34.6236944Z              results.push_back({ node, static_cast<int64_t>(color + 1) });
2021-11-12T16:01:34.6237579Z                                  ^~~~
2021-11-12T16:01:34.6238048Z                                  {   }
2021-11-12T16:01:34.6239571Z /home/runner/work/pgrouting/pgrouting/include/coloring/pgr_sequentialVertexColoring.hpp:135:40: warning: suggest braces around initialization of subobject [-Wmissing-braces]
2021-11-12T16:01:34.6240941Z              results.push_back({ node, static_cast<int64_t>(color + 1) });
2021-11-12T16:01:34.6241588Z                                        ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
2021-11-12T16:01:34.6242073Z                                        {                              }
2021-11-12T16:01:35.3308928Z /home/runner/work/pgrouting/pgrouting/include/coloring/pgr_sequentialVertexColoring.hpp:135:34: warning: suggest braces around initialization of subobject [-Wmissing-braces]
2021-11-12T16:01:35.3310794Z              results.push_back({ node, static_cast<int64_t>(color + 1) });
2021-11-12T16:01:35.3311510Z                                  ^~~~
2021-11-12T16:01:35.3312308Z                                  {   }
2021-11-12T16:01:35.3315799Z /home/runner/work/pgrouting/pgrouting/include/coloring/pgr_sequentialVertexColoring.hpp:108:20: note: in instantiation of member function 'pgrouting::functions::Pgr_sequentialVertexColoring<pgrouting::graph::Pgr_base_graph<boost::adjacency_list<boost::vecS, boost::vecS, boost::undirectedS, pgrouting::Basic_vertex, pgrouting::Basic_edge, boost::no_property, boost::listS>, pgrouting::Basic_vertex, pgrouting::Basic_edge> >::get_results' requested here
2021-11-12T16:01:35.3318161Z          results = get_results(colors, graph);
2021-11-12T16:01:35.3318878Z                    ^
2021-11-12T16:01:35.3322476Z /home/runner/work/pgrouting/pgrouting/src/coloring/sequentialVertexColoring_driver.cpp:64:48: note: in instantiation of member function 'pgrouting::functions::Pgr_sequentialVertexColoring<pgrouting::graph::Pgr_base_graph<boost::adjacency_list<boost::vecS, boost::vecS, boost::undirectedS, pgrouting::Basic_vertex, pgrouting::Basic_edge, boost::no_property, boost::listS>, pgrouting::Basic_vertex, pgrouting::Basic_edge> >::sequentialVertexColoring' requested here
2021-11-12T16:01:35.3326062Z     auto results = fn_sequentialVertexColoring.sequentialVertexColoring(graph);
2021-11-12T16:01:35.3327232Z                                                ^
2021-11-12T16:01:35.3331342Z /home/runner/work/pgrouting/pgrouting/src/coloring/sequentialVertexColoring_driver.cpp:120:19: note: in instantiation of function template specialization 'pgr_sequentialVertexColoring<pgrouting::graph::Pgr_base_graph<boost::adjacency_list<boost::vecS, boost::vecS, boost::undirectedS, pgrouting::Basic_vertex, pgrouting::Basic_edge, boost::no_property, boost::listS>, pgrouting::Basic_vertex, pgrouting::Basic_edge> >' requested here
2021-11-12T16:01:35.3335304Z         results = pgr_sequentialVertexColoring(undigraph);
2021-11-12T16:01:35.3336071Z                   ^
2021-11-12T16:01:35.3337106Z In file included from /home/runner/work/pgrouting/pgrouting/src/coloring/sequentialVertexColoring_driver.cpp:39:
2021-11-12T16:01:35.3339861Z /home/runner/work/pgrouting/pgrouting/include/coloring/pgr_sequentialVertexColoring.hpp:135:40: warning: suggest braces around initialization of subobject [-Wmissing-braces]
2021-11-12T16:01:35.3341774Z              results.push_back({ node, static_cast<int64_t>(color + 1) });
2021-11-12T16:01:35.3342451Z                                        ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
2021-11-12T16:01:35.3342984Z                                        {                              }
2021-11-12T16:01:36.3154069Z 4 warnings generated.
2021-11-12T16:01:36.3526078Z [ 84%] Building C object src/coloring/CMakeFiles/coloring.dir/bipartite.c.o
2021-11-12T16:01:36.4241785Z [ 85%] Building C object src/breadthFirstSearch/CMakeFiles/breadthFirstSearch.dir/binaryBreadthFirstSearch.c.o
2021-11-12T16:01:36.6776013Z [ 85%] Building CXX object src/coloring/CMakeFiles/coloring.dir/bipartite_driver.cpp.o
2021-11-12T16:01:36.7568803Z [ 85%] Building CXX object src/breadthFirstSearch/CMakeFiles/breadthFirstSearch.dir/binaryBreadthFirstSearch_driver.cpp.o
2021-11-12T16:01:41.9322247Z [ 99%] Building CXX object src/coloring/CMakeFiles/coloring.dir/pgr_edgeColoring.cpp.o
2021-11-12T16:01:42.3340207Z In file included from /home/runner/work/pgrouting/pgrouting/src/coloring/bipartite_driver.cpp:46:
2021-11-12T16:01:42.3468583Z /home/runner/work/pgrouting/pgrouting/include/coloring/pgr_bipartite_driver.hpp:81:44: warning: suggest braces around initialization of subobject [-Wmissing-braces]
2021-11-12T16:01:42.3470313Z                         results.push_back({vid, 0}) :results.push_back({vid, 1});
2021-11-12T16:01:42.3471293Z                                            ^~~
2021-11-12T16:01:42.3472087Z                                            {  }
2021-11-12T16:01:42.3473632Z /home/runner/work/pgrouting/pgrouting/include/coloring/pgr_bipartite_driver.hpp:81:49: warning: suggest braces around initialization of subobject [-Wmissing-braces]
2021-11-12T16:01:42.3475089Z                         results.push_back({vid, 0}) :results.push_back({vid, 1});
2021-11-12T16:01:42.3475598Z                                                 ^
2021-11-12T16:01:42.3475926Z                                                 {}
2021-11-12T16:01:42.3477239Z /home/runner/work/pgrouting/pgrouting/include/coloring/pgr_bipartite_driver.hpp:81:73: warning: suggest braces around initialization of subobject [-Wmissing-braces]
2021-11-12T16:01:42.3478741Z                         results.push_back({vid, 0}) :results.push_back({vid, 1});
2021-11-12T16:01:42.3479258Z                                                                         ^~~
2021-11-12T16:01:42.3479644Z                                                                         {  }
2021-11-12T16:01:42.3481063Z /home/runner/work/pgrouting/pgrouting/include/coloring/pgr_bipartite_driver.hpp:81:78: warning: suggest braces around initialization of subobject [-Wmissing-braces]
2021-11-12T16:01:42.3482136Z                         results.push_back({vid, 0}) :results.push_back({vid, 1});
2021-11-12T16:01:42.3482631Z                                                                              ^
2021-11-12T16:01:42.3483019Z                                                                              {}
2021-11-12T16:01:43.3088411Z /home/runner/work/pgrouting/pgrouting/include/coloring/pgr_bipartite_driver.hpp:81:44: warning: suggest braces around initialization of subobject [-Wmissing-braces]
2021-11-12T16:01:43.3089738Z                         results.push_back({vid, 0}) :results.push_back({vid, 1});
2021-11-12T16:01:43.3090302Z                                            ^~~
2021-11-12T16:01:43.3090662Z                                            {  }
2021-11-12T16:01:43.3094767Z /home/runner/work/pgrouting/pgrouting/include/coloring/pgr_bipartite_driver.hpp:89:42: note: in instantiation of member function 'pgrouting::functions::Pgr_Bipartite<pgrouting::graph::Pgr_base_graph<boost::adjacency_list<boost::vecS, boost::vecS, boost::undirectedS, pgrouting::Basic_vertex, pgrouting::Basic_edge, boost::no_property, boost::listS>, pgrouting::Basic_vertex, pgrouting::Basic_edge> >::print_Bipartite' requested here
2021-11-12T16:01:43.3098881Z                 if (bipartite) results = print_Bipartite(graph);
2021-11-12T16:01:43.3099400Z                                          ^
2021-11-12T16:01:43.3102443Z /home/runner/work/pgrouting/pgrouting/src/coloring/bipartite_driver.cpp:80:32: note: in instantiation of member function 'pgrouting::functions::Pgr_Bipartite<pgrouting::graph::Pgr_base_graph<boost::adjacency_list<boost::vecS, boost::vecS, boost::undirectedS, pgrouting::Basic_vertex, pgrouting::Basic_edge, boost::no_property, boost::listS>, pgrouting::Basic_vertex, pgrouting::Basic_edge> >::pgr_bipartite' requested here
2021-11-12T16:01:43.3106542Z         results = fn_Bipartite.pgr_bipartite(undigraph);
2021-11-12T16:01:43.3108019Z                                ^
2021-11-12T16:01:43.3111803Z In file included from /home/runner/work/pgrouting/pgrouting/src/coloring/bipartite_driver.cpp:46:
2021-11-12T16:01:43.3116992Z /home/runner/work/pgrouting/pgrouting/include/coloring/pgr_bipartite_driver.hpp:81:49: warning: suggest braces around initialization of subobject [-Wmissing-braces]
2021-11-12T16:01:43.3120782Z                         results.push_back({vid, 0}) :results.push_back({vid, 1});
2021-11-12T16:01:43.3123490Z                                                 ^
2021-11-12T16:01:43.3126028Z                                                 {}
2021-11-12T16:01:43.3131368Z /home/runner/work/pgrouting/pgrouting/include/coloring/pgr_bipartite_driver.hpp:81:73: warning: suggest braces around initialization of subobject [-Wmissing-braces]
2021-11-12T16:01:43.3135013Z                         results.push_back({vid, 0}) :results.push_back({vid, 1});
2021-11-12T16:01:43.3137695Z                                                                         ^~~
2021-11-12T16:01:43.3140211Z                                                                         {  }
2021-11-12T16:01:43.3145117Z /home/runner/work/pgrouting/pgrouting/include/coloring/pgr_bipartite_driver.hpp:81:78: warning: suggest braces around initialization of subobject [-Wmissing-braces]
2021-11-12T16:01:43.3152306Z                         results.push_back({vid, 0}) :results.push_back({vid, 1});
2021-11-12T16:01:43.3154931Z                                                                              ^
2021-11-12T16:01:43.3157279Z                                                                              {}
2021-11-12T16:01:44.3720167Z 8 warnings generated.
2021-11-12T16:01:44.9930057Z [ 99%] Built target breadthFirstSearch
2021-11-12T16:01:45.6371511Z /home/runner/work/pgrouting/pgrouting/src/coloring/pgr_edgeColoring.cpp:62:28: warning: suggest braces around initialization of subobject [-Wmissing-braces]
2021-11-12T16:01:45.6372627Z         results.push_back({edge, (color + 1)});
2021-11-12T16:01:45.6373446Z                            ^~~~
2021-11-12T16:01:45.6373822Z                            {   }
2021-11-12T16:01:45.6375052Z /home/runner/work/pgrouting/pgrouting/src/coloring/pgr_edgeColoring.cpp:62:34: warning: suggest braces around initialization of subobject [-Wmissing-braces]
2021-11-12T16:01:45.6376075Z         results.push_back({edge, (color + 1)});
2021-11-12T16:01:45.6376509Z                                  ^~~~~~~~~~~
2021-11-12T16:01:45.6376883Z                                  {          }
2021-11-12T16:01:46.3580240Z 2 warnings generated.
2021-11-12T16:01:46.3753713Z [ 99%] Built target coloring
2021-11-12T16:01:46.3868192Z [100%] Linking CXX shared library lib/libpgrouting-3.3.so
2021-11-12T16:01:50.5127936Z [100%] Built target pgrouting-3.3
```

--- a/NEWS
+++ b/NEWS
@@ -40,6 +40,17 @@ pgRouting 3.4.0 Release Notes
   * ``pgr_trspViaEdges(text,integer[],float[],boolean,boolean,text)``
 
 
+pgRouting 3.3.1 Release Notes
+-------------------------------------------------------------------------------
+
+To see all issues & pull requests closed by this release see the `Git closed
+milestone for 3.3.1
+<https://github.com/pgRouting/pgrouting/issues?utf8=%E2%9C%93&q=milestone%3A%22Release%203.3.1%22>`_
+on Github.
+
+**Issue fixes**
+
+* [#2216](https://github.com/pgRouting/pgrouting/issues/2216): Warnings when using clang
 
 pgRouting 3.3.0 Release Notes
 -------------------------------------------------------------------------------

--- a/doc/src/release_notes.rst
+++ b/doc/src/release_notes.rst
@@ -33,7 +33,6 @@ To see the full list of changes check the list of `Git commits <https://github.c
    :local:
 
 
-
 pgRouting 3.4.0 Release Notes
 -------------------------------------------------------------------------------
 
@@ -76,6 +75,17 @@ pgRouting 3.4.0 Release Notes
   * ``pgr_trspViaEdges(text,integer[],float[],boolean,boolean,text)``
 
 
+pgRouting 3.3.1 Release Notes
+-------------------------------------------------------------------------------
+
+To see all issues & pull requests closed by this release see the `Git closed
+milestone for 3.3.1
+<https://github.com/pgRouting/pgrouting/issues?utf8=%E2%9C%93&q=milestone%3A%22Release%203.3.1%22>`_
+on Github.
+
+.. rubric:: Issue fixes
+
+* `#2216 <https://github.com/pgRouting/pgrouting/issues/2216>`__: Warnings when using clang
 
 pgRouting 3.3.0 Release Notes
 -------------------------------------------------------------------------------

--- a/include/coloring/pgr_bipartite_driver.hpp
+++ b/include/coloring/pgr_bipartite_driver.hpp
@@ -78,7 +78,7 @@ class Pgr_Bipartite : public pgrouting::Pgr_messages {
                     int64_t vid = graph[*v].id;
                     boost::get(partition_map, *v) ==
                         boost::color_traits <boost::default_color_type>::white() ?
-                        results.push_back({vid, 0}) :results.push_back({vid, 1});
+                        results.push_back({{vid}, {0}}) :results.push_back({{vid}, {1}});
                 }
                 return results;
             }

--- a/include/coloring/pgr_sequentialVertexColoring.hpp
+++ b/include/coloring/pgr_sequentialVertexColoring.hpp
@@ -132,7 +132,7 @@ class Pgr_sequentialVertexColoring {
          for (boost::tie(v, vend) = vertices(graph.graph); v != vend; ++v) {
              int64_t node = graph[*v].id;
              auto color = colors[*v];
-             results.push_back({ node, static_cast<int64_t>(color + 1) });
+             results.push_back({{node}, {static_cast<int64_t>(color + 1)}});
          }
 
          // ordering the results in an increasing order of the node id

--- a/include/components/pgr_makeConnected.hpp
+++ b/include/components/pgr_makeConnected.hpp
@@ -88,7 +88,7 @@ class Pgr_makeConnected : public pgrouting::Pgr_messages {
               int64_t tgt = graph[graph.target(*ei)].id;
               log<< "src:" << src<< "tgt:" << tgt <<"\n";
               if (newEdge >= edgeCount) {
-                   results[i] = {src, tgt};
+                   results[i] = {{src}, {tgt}};
                    i++;
               }
               newEdge++;

--- a/include/dominator/pgr_lengauerTarjanDominatorTree_driver.hpp
+++ b/include/dominator/pgr_lengauerTarjanDominatorTree_driver.hpp
@@ -82,7 +82,7 @@ class Pgr_LTDTree : public pgrouting::Pgr_messages {
                 V_i v, vend;
                 for (boost::tie(v, vend) = vertices(graph.graph); v != vend; ++v) {
                     int64_t vid = graph[*v].id;
-                    results.push_back({vid, (idoms[*v] != -1 ? (idoms[*v]+1) : 0) });
+                    results.push_back({{vid}, {(idoms[*v] != -1 ? (idoms[*v]+1) : 0)}});
                 }
 
                  return results;

--- a/src/bdAstar/bdAstar_driver.cpp
+++ b/src/bdAstar/bdAstar_driver.cpp
@@ -108,7 +108,7 @@ pgr_bdAstar(
                     return lhs.d1.source < rhs.d1.source;
                 });
 
-        II_t_rt previousCombination{0, 0};
+        II_t_rt previousCombination {{0}, {0}};
 
         for (const II_t_rt &comb : combinations) {
             fn_bdAstar.clear();

--- a/src/bdDijkstra/bdDijkstra_driver.cpp
+++ b/src/bdDijkstra/bdDijkstra_driver.cpp
@@ -97,7 +97,7 @@ pgr_bdDijkstra(
                     return lhs.d1.source < rhs.d1.source;
                 });
 
-        II_t_rt previousCombination{0, 0};
+        II_t_rt previousCombination {{0}, {0}};
 
         for (const II_t_rt &comb : combinations) {
             fn_bdDijkstra.clear();

--- a/src/coloring/pgr_edgeColoring.cpp
+++ b/src/coloring/pgr_edgeColoring.cpp
@@ -59,7 +59,7 @@ Pgr_edgeColoring::edgeColoring() {
     for (auto e_i : boost::make_iterator_range(boost::edges(graph))) {
         auto edge = get_edge_id(e_i);
         int64_t color = graph[e_i];
-        results.push_back({edge, (color + 1)});
+        results.push_back({{edge}, {(color + 1)}});
     }
     return results;
 }

--- a/src/components/componentsResult.cpp
+++ b/src/components/componentsResult.cpp
@@ -48,7 +48,7 @@ componentsResult(
     for (const auto& component : components) {
         auto component_id = component[0];
         for (const auto edge_id : component) {
-            results.push_back({edge_id, component_id});
+            results.push_back({{edge_id}, {component_id}});
         }
     }
     return results;

--- a/src/trsp/pgr_trspHandler.cpp
+++ b/src/trsp/pgr_trspHandler.cpp
@@ -353,7 +353,7 @@ Pgr_trspHandler::process(
         const std::map<int64_t, std::set<int64_t>> &combinations
         ) {
     std::deque<Path> paths;
-    for (const auto c : combinations) {
+    for (const auto &c : combinations) {
         for (const auto target : c.second) {
             paths.push_back(process(c.first, target));
         }

